java/vector/src/main/java/org/apache/arrow/vector/complex/FixedSizeListVector.java

```diff
@@ -579,10 +579,74 @@ public void transfer() {
 
     @Override
     public void splitAndTransfer(int startIndex, int length) {
+      assert startIndex + length <= valueCount;
```

Comment:
Assertions should be used for checking arguments, only internal concerns. This is failed assertions produce Java Errors which are not caught by traditional catch clauses. Please use something akin to Preconditions.

Reply:
Thanks for pointing, fixed.